editor/src/components/canvas/controls/comment-popup.tsx

```diff
@@ -225,6 +249,27 @@ const CommentThread = React.memo(({ comment }: CommentThreadProps) => {
 
   const collabs = useStorage((storage) => storage.collaborators)
 
+  const onScroll = () => {
+    const element = listRef.current
+    if (element == null) {
+      return
+    }
+    const tolerance = 20 // px
+
+    const atBottom = element.scrollHeight - element.scrollTop <= PopupMaxHeight + tolerance
+    const isOverflowing = element.scrollHeight > PopupMaxHeight
+    setShowShadowBottom(!atBottom && isOverflowing)
+
+    const atTop = element.scrollTop > tolerance
+    setShowShadowTop(atTop && isOverflowing)
+  }
+
+  React.useEffect(() => {
+    // when the thread id changes, re-check the scroll and set the inset shadow
+    onScroll()
+    scrollToBottom()
+  }, [threadId, scrollToBottom])
```

Comment:
Is this necessary? As far as I know the threadId can only change when we are creating a new thread by typing the first comment, and in that case we don't have to scroll (there is only a single comment in the thread).

Reply:
But why? Does it run only once at the beginning, and that is needed?

Reply:
if you have an open comment and click on another comment, the thread id changes and the scroll needs to change as well, but without the useeffect the component is not destroyed and the scroll stuff (e.g. shadows) does not change

Reply:
Ok, I did not realize that the same element without unmount/remount can be switched to a different comment thread.